extensions/kubectl-cli/src/extension.ts

```diff
@@ -362,6 +363,30 @@ async function postActivate(
       releaseVersionToInstall = undefined;
       releaseToInstall = undefined;
     },
+    doUninstall: async _logger => {
+      if (!currentVersion) {
+        throw new Error(`Cannot uninstall ${kubectlCliName}. No version detected.`);
+      }
+
+      // delete the executable stored in the storage folder
+      const storagePath = getStorageKubectlPath(extensionContext);
+      if (fs.existsSync(storagePath)) {
+        await fs.promises.unlink(storagePath);
+      }
+
+      // delete the executable in the system path
+      const systemPath = getSystemBinaryPath(kubectlCliName);
+      if (fs.existsSync(systemPath)) {
+        await fs.promises.unlink(systemPath);
+      }
+
+      // update the version to undefined
+      currentVersion = undefined;
+      kubectlCliTool?.updateVersion({
+        version: undefined,
+        path: undefined,
+      });
```

Comment:
here it's not clear to me, should we have a remove or uninstall method instead ?

Reply:
by thinking at it maybe it should not be the extension to update the version. When the core calls the uninstall and succeed then it knows that the version is not there anymore and updates it? So no need for a remove/uninstall method?

Reply:
maybe it may look more natural the way you describe

but then there should be event fired by the framework (like onDidUninstall)

Reply:
sure 👍

Reply:
updated